streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -296,6 +296,75 @@ public void shouldAddTasksToStateUpdater() {
         Mockito.verify(stateUpdater).add(task01);
     }
 
+    @Test
+    public void shouldHandleRemovedTasksFromStateUpdater() {
```

Comment:
We also need a test to verify that new tasks are initialized and added to the state updater.

Reply:
I just saw that `shouldAddTasksToStateUpdater()` is such a test. The verification for calling `initializeIfNeeded()` is missing, though.

Reply:
This test did verify that `initializeIfNeeded` is called, and the tasks are added back to the state updater.